Dsikora369_code/task_1/titanic.py

```diff
@@ -0,0 +1,286 @@
+# -*- coding: utf-8 -*-
+"""Titanic.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/11X_XykBliOu0vOzxkhNBvPBXDekZWxgu
+
+## Titanic competition Sikorskyi ##
+
+**Downloading datasets from kaggle**
+
+Using Google Colabs functions download Kaggle token.
+"""
+
+import pandas as pd
+import numpy as np
+import re
+from google.colab import files
+from sklearn.preprocessing import LabelEncoder
+from sklearn.model_selection import train_test_split
+import tensorflow as tf
+import math
+from tensorflow.keras.layers import Dense, Dropout
+files.upload()  # here you will download kaggle.json
+
+"""Set permission to before downloading Titanic dataset."""
+
+
+"""Downloading dataset.
+
+"""
+
+"""Getting our data.
+
+"""
+
+train = pd.read_csv('train.csv')
+test = pd.read_csv('test.csv')
+train
+
+"""**Data transformation**
+
+Befor creating model, it is necessary to analyze, then create function to clean data.
+
+Sibsp and Parch refer to family, so it is logical to replace them with one column and create 'Alone' column for people without family members.
+"""
+
+train['FamilySize'] = train['SibSp'] + train['Parch'] + 1
+train[['FamilySize', 'Survived']].groupby(['FamilySize'],
+                                          as_index=False).mean()
+
+train['Alone'] = 0
+train.loc[train['FamilySize'] == 1, 'Alone'] = 1
+train[['Alone', 'Survived']].groupby(['Alone'], as_index=False).mean()
+
+"""'Fare' has NaNs, so they will be replaced to mean. Also, for future model it is better to replace 'Fare' with its categorical representation. """
+
+train['Fare'] = train['Fare'].fillna(train['Fare'].median())
+train['CategoricalFare'] = pd.qcut(train['Fare'], 5)
+train[['CategoricalFare', 'Survived']].groupby(
+    ['CategoricalFare'], as_index=False).mean()
+
+"""Also, it is very useful to notice whether passanger had a cabin or not and which cabin.
+
+"""
+
+
+def get_cabin(cabin):
+    for i in str(cabin):
+        if i in ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'Z']:

```

Comment:
```suggestion
        if i in 'ABCDEFGHZ':

```

Python rows is arrays on itself, so you may not split them letter-by-letter